packages/uicore/src/components/StringSubstitute/StringSubstitute.tsx

```diff
@@ -1,49 +1,63 @@
 import React, { Fragment } from 'react'
 
 type SubstituteVars = Record<string, any> // eslint-disable-line @typescript-eslint/no-explicit-any
+const MATCH_ELSE_KEY = '___'
+const MAX_LOOP_COUNT = 1000
```

Comment:
*Minor* the name might be a bit to generic - maybe it could be sth more specific like `MAX_EXPRESSION_TRANSLATIONS_PER_KEY`